src/abstractdataframe/io.jl

```diff
@@ -164,128 +165,151 @@ function html_escape(cell::AbstractString)
+        mxcol = -1
     end
-    write(io, "</thead>")
-    write(io, "<tbody>")
-    for row in 1:mxrow
-        write(io, "<tr>")
-        if rowid === nothing
-            write(io, "<th>$row</th>")
-        else
-            write(io, "<th>$rowid</th>")
-        end
-        for column_name in cnames
-            if isassigned(df[!, column_name], row)
-                cell_val = df[row, column_name]
-                if ismissing(cell_val)
-                    write(io, "<td><em>missing</em></td>")
-                elseif cell_val isa Markdown.MD
-                    write(io, "<td>")
-                    show(io, "text/html", cell_val)
-                    write(io, "</td>")
-                elseif cell_val isa SHOW_TABULAR_TYPES
-                    write(io, "<td><em>")
-                    cell = sprint(ourshow, cell_val, 0)
-                    write(io, html_escape(cell))
-                    write(io, "</em></td>")
-                else
-                    cell = sprint(ourshow, cell_val, 0)
-                    write(io, "<td>$(html_escape(cell))</td>")
-                end
-            else
-                write(io, "<td><em>#undef</em></td>")
-            end
+
+    # Check if the user wants to display a summary about the DataFrame that is
+    # being printed. This will be shown using the `title` option of
+    # `pretty_table`.
+    if summary
+        if isempty(title)
+            title = Base.summary(df)
         end
-        write(io, "</tr>")
+    else
+        title = ""
     end
-    if size(df, 1) > mxrow
-        write(io, "<tr>")
-        write(io, "<th>&vellip;</th>")
-        for column_name in cnames
-            write(io, "<td>&vellip;</td>")
+
+    # If `rowid` is not `nothing`, then we are printing a data row. In this
+    # case, we will add this information using the row name column of
+    # PrettyTables.jl. Otherwise, we can just use the row number column.
+    if (rowid === nothing) || (ncol(df) == 0)
+        show_row_number::Bool = get(kwargs, :show_row_number, true)
+        row_labels = nothing
+
+        # If the columns with row numbers is not shown, then we should not
+        # display a vertical line after the first column.
+        vlines = fill(1, show_row_number)
+    else
+        nrow(df) != 1 &&
+            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+
+        # In this case, if the user does not want to show the row number, then
+        # we must hide the row name column, which is used to display the
+        # `rowid`.
+        if !get(kwargs, :show_row_number, true)
+            row_labels = nothing
+            vlines = Int[]
+        else
+            row_labels = [string(rowid)]
+            vlines = Int[1]
         end
-        write(io, "</tr>")
+
+        show_row_number = false
     end
-    write(io, "</tbody>")
-    write(io, "</table>")
-    write(io, "</div>")
+
+    maximum_columns_width = truncate <= 0 ? "" : string(truncate) * "px"
```

Comment:
Maybe we should just let users pass this string directly (rather than an integer), so that we can support different units in the future if needed? Pixels only make sense for some backends, one could also want to truncate at a number of columns. Actually, why not pass `maximum_columns_width` directly?

Reply:
I think we should keep `truncate` to make the keyword works in all backends (text and HTML). Hence, the user will not get error when the type of output changes. However, notice that the user can still pass the string directly, which will have higher precedence than the one we create inside this function:

<img width="873" alt="Captura de Tela 2022-09-19 às 10 46 25" src="https://user-images.githubusercontent.com/1068295/191032150-c2bf6ec8-133b-4fd4-a6e5-7fefb3ccaa50.png">

Reply:
If several backends accept `truncate`, then it should have the same meaning everywhere. Otherwise it's not really useful (and potentially misleading), especially since pixels and numbers of characters give completely different results.

Reply:
So, do we remover this option from HTML?

Reply:
@bkamins What do you think?

Reply:
We need this option as in html columns can get super wide otherwise (e.g. if someone stores large objects in a columns). We have two ways to solve it:
* use a different kwarg name in HTML, e.g. `truncatepx`
* use the same kwarg, but clearly document the difference (in practice I do not think anyone would be confused by this)

Reply:
What would be the drawback of requiring users to pass`"5px"` as a string instead of a raw number?

Reply:
No drawback - it is one way of handling option 2 essentially. As it would need to be documented that in text/plain a number is passed and in HTML `px` has to be suffixed. For now I would check if the format of passed value is one or more digits+suffixed by "px" if we go this way.

Reply:
Should we change the meaning of this keyword? What is the decision here?

Reply:
We are still discussing 😄.

I have looked at the issue more closely. Let me summarize what we already have:
* plain: `truncate` sets number of character width to display
* HTML: `max_column_width`: column width in pixels

Therefore I propose to support passing `truncate` in HTML mode but ignore it. And document that users should pass `max_column_width` instead. @nalimilan - what do you think?

Reply:
Is there a particular reason to think that users will want to pass `truncate` in a context where the backend may be either text or HTML? I find it potentially confusing to silently ignore a keyword argument. I can't count the number of times this made absurd calls return a misleading result when programming in R.

Reply:
The reason is that you have code written for text (without thinking about the fact that later someone might want to run it under Jupyter Notebook) and just copy-pasting it to HTML.

However, indeed it is true also in reverse so if we have `max_column_width` in HTML and someone copies this code to text it will error. So maybe it is OK not to support `truncate` in HTML and error.

In that case what we should do maybe is:
* in text: print informative error message that `max_column_width` is not supported (if someone uses it) in text and prompt user to change it to `truncate`
* in HTML: print informative error message that `truncate` is not supported (if someone uses it) in HTML and prompt user to change it to `max_column_width`

The general idea is that kwargs supported in one type of output should print **informative** error message in other outputs so that the code is easily fixable and user is aware of the situation.

Reply:
Ah so if you call `show(df, ...)` under Jupyter it will automatically use HTML? (Sorry, I don't use it.)

Reply:
Yes - this is the point. (later a section to the manual will be added by @ronisbr clarifying these things, but first we want the PR done)

@ronisbr BTW in
![image](https://user-images.githubusercontent.com/6187170/191208433-12db822c-44ae-4f4b-9f23-b7706e9d2624.png)
why there is a difference in summary (it has different font)

Reply:
@ronisbr - now I actually see the reason. Sorry for mixing this. So when you use `show` explicitly in JupyterNotebook then text/plain is used, so it is not a problem. You have to use `show(stdout, MIME("text/html"), DataFrame(a=1:2))` to use HTML output.

@nalimilan - so you now have above all information how things work.

Reply:
OK, so Jupyter doesn't switch the output to HTML automatically. Then I'd say that throwing errors as you proposed above at https://github.com/JuliaData/DataFrames.jl/pull/3096#discussion_r975033841 is the best solution. (As usual, the advantage with errors is that we can turn them into something else later if we want, while the contrary isn't possible.)